block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -133,27 +127,23 @@ impl BlockWitness {
 
         let accounts_in_batches: BTreeSet<AccountId> =
             batches_initial_states.keys().cloned().collect();
-        let accounts_in_store: BTreeSet<AccountId> = block_inputs
-            .account_states
-            .iter()
-            .map(|record| &record.account_id)
-            .cloned()
-            .collect();
+        let accounts_in_store: BTreeSet<AccountId> =
+            block_inputs.accounts.iter().map(|(&account_id, _)| account_id).collect();
```

Comment:
Maybe use the `keys` API?

```suggestion
            block_inputs.accounts.keys().collect();
```